.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,12 +1,22 @@
-<!--
-Please follow the naming conventions in the "guidelines for contributing" link above or the pull request may be rejected.
+Fixes #
 
-Also, do consider outlining the solution taken as doing so will likely help in the reviewing process
-(e.g., when the pull request involves non-trivial changes)
--->
+**PR Checklist**
 
-Fixes #
+<!-- Remove this portion after you have made the checks. -->
+
+Have you:
+- [ ] Read and understood [our PR guideline (steps 4-6)](https://github.com/TEAMMATES/teammates/blob/master/docs/process.md#step-4-submit-a-pr)?
+  - [ ] Added the issue number to the "Fixes" keyword above?
+  - [ ] Titled the PR as specified in the abovementioned document?
+- [ ] Made your changes on a branch other than `master`?
```

Comment:
Shall we have a stronger requirement on this one? Like actually specify exactly what the branch name should be?

Otherwise, someone may open a PR from `develop` or a similarly disturbing branch name.